lib/client.c

```diff
@@ -18,6 +18,7 @@
 #include "chunk_downloader.h"
 
 #define curl_easier_escape(curl, string) curl_easy_escape(curl, string, 0)
+#define MAX_NAME_SIZE 20
```

Comment:
20 is almost definitely too small. I would say at least 256 is your minimum here. There is no max size according to the JDBC standard (or at least I can't find any place that states a max size).